x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/TransportDownsampleAction.java

```diff
@@ -286,143 +297,205 @@ protected void masterOperation(
-                                    },
-                                        e -> deleteRollupIndex(
-                                            sourceIndexName,
-                                            rollupIndexName,
-                                            parentTask,
-                                            listener,
-                                            new ElasticsearchException("Failed to refresh rollup index [" + rollupIndexName + "]", e)
-                                        )
-                                    ));
-                                } else {
-                                    deleteRollupIndex(
-                                        sourceIndexName,
-                                        rollupIndexName,
-                                        parentTask,
-                                        listener,
-                                        new ElasticsearchException("Unable to update settings of rollup index [" + rollupIndexName + "]")
-                                    );
+                                    }
+                                ),
+                                e -> {
+                                    if (e instanceof ResourceAlreadyExistsException) {
+                                        waitForExistingTask(request, listener, persistentRollupTaskId);
+                                    } else {
+                                        listener.onFailure(e);
+                                    }
                                 }
-                            },
-                                e -> deleteRollupIndex(
-                                    sourceIndexName,
-                                    rollupIndexName,
-                                    parentTask,
-                                    listener,
-                                    new ElasticsearchException("Unable to update settings of rollup index [" + rollupIndexName + "]", e)
-                                )
-                            ));
-                        } else {
-                            deleteRollupIndex(
-                                sourceIndexName,
-                                rollupIndexName,
-                                parentTask,
-                                listener,
-                                new ElasticsearchException("Unable to index into rollup index [" + rollupIndexName + "]")
-                            );
-                        }
-                    }, e -> deleteRollupIndex(sourceIndexName, rollupIndexName, parentTask, listener, e)));
+                            )
+                        );
+                    }
                 } else {
                     listener.onFailure(new ElasticsearchException("Failed to create rollup index [" + rollupIndexName + "]"));
                 }
             }, listener::onFailure));
         }, listener::onFailure));
     }
 
+    private void waitForExistingTask(
```

Comment:
Maybe we should return a different response here?
Downsampling in progress with a task id that can be used in get task api to see the progress?
If no we should just delegate to `listener` on line 462.